src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -164,6 +164,26 @@ public void updateFilteredPatientList(Predicate<Patient> predicate) {
         filteredPatients.setPredicate(predicate);
     }
 
+    @Override
+    public Index getPatientIndex(Name patientName) {
```

Comment:
Would it not be possible to use `getPatientWithName` instead? I feel like that may significantly reduce the amount of code that we need.

Reply:
not really, kinda forced to do it this way unless i dont use `Index` class for this method

Reply:
I think the patient name is the identifier for a patient right? In that case, wouldn't it better to avoid using the `Index` class?

Reply:
if i dont use `Index` class right, then i'll be using `integer` instead. cuz if i use `roomlist.getObservableList.indexOf(getPatientWithName("some name here")`, it will return `-1` if the patient is not found. but `Index` class doesnt accept negative integer, so it will throw out an exception

Reply:
so its kinda like a comparison of using int vs using Index, because either way, i need the position of the patient from the list

Reply:
If I am not wrong, the two methods `getPatientIndex()` and `getPatientFromIndex()` are used in the `EditPatientCommand` class, right?

Instead of this

```
        Index index = model.getPatientIndex(patientToBeEdited);

        if (index.getZeroBased() == 0) {
            throw new CommandException(Messages.MESSAGE_INVALID_PATIENT_NAME_INPUT);
        }

        Patient patientToEdit = model.getPatientFromIndex(Index.fromZeroBased(index.getZeroBased() - 1));
```

Would it not be cleaner to do this in `EditPatientCommand`?

```
        Optional<Patient> optionalPatient  = model.getPatientWithName(patientToBeEdited);

        // Better still, use optionalPatient.orElseThrow(...). I forgot the syntax hahaha.
        if (!optionalPatient.isPresent()) {
            throw new CommandException(Messages.MESSAGE_INVALID_PATIENT_NAME_INPUT);
        }

        Patient patientToEdit = optionalPatient.get();
```

That way, we avoid using Index altogether.

Reply:
yup, wasnt really thinking sorz ><